### DISCUSSION TITLE
Make Flatpak release

### DISCUSSION THREAD

**Discussion by blacklightpy** [kimlimjustin/xplorer#304] [General]
AppImage won't work with musl libc. Flatpak solves the problem by including the runtime just like containers do. AppImages currently still depend on glibc (https://github.com/AppImage/AppImageKit/issues/877).

**Comment by curtisy1** [comment]
While I agree that Flatpaks would be a better approach for a multitude of reasons, note that this won't be possible until Tauri 2 is released (or used in this project) as far as I know. The Tauri team briefly mentioned Flatpak support [here](https://tauri.app/blog/2023/02/03/tauri-2-0-0-alpha-3/#will-this-bring-breaking-changes-to-my-code) although I'm not sure if that idea ever progressed further.

The quickest way would be submitting a request to include it in Flathub like [Cinny did](https://github.com/flathub/in.cinny.Cinny). 
If you can give me a rundown of what I need to do to set this up or send me a link that I can follow, I'll see what I can do

**Comment by blacklightpy** [comment]
@curtisy1 I've had trouble packaging a game into Flatpak, but probably required Mesa and stuff, and I didn't spend more than 30 minutes trying to learn it.

Here is a simply Flatpak Hello World: https://docs.flatpak.org/en/latest/first-build.html

Basically you have to add the files and add in the runtime, and it should work. The manifest is about all that you have to create, and flatpak builder will create the Flatpak installation. It would be just like running a binary in an OCI container instance.

You can refer Cinny's manifest, which uses the Freedesktop `node20` and `rust-stable` SDK extensions, and no runtime extensions. It builds two programs, Cinny and libvips, which is built with Meson. Cinny is built with the simple build system, which is just manual commands as you can see. The binaries are then moved to `/app/bin`, which is the default spot for Flatpak apps similar to `/usr/bin`, but both are in the PATH variable.

The source files are copied in to the `/app` folder, and can be of many types such as `file`, `dir`, `git` and `patch`. I didn't read about the patch type, but looking at Cinny's manifest, I think patch is applied when it is copied. I think the SDK extensions handle installing from the `node-sources.json` and `cargo-sources.json` files.

I don't know why webkit2gtk-4.0 is a module, and it further has modules in its section, when libvips is not a module. But once you figure that out, I guess copying the binary would be trivial.

I didn't have time to spend learning it, but someone else's who's made a Flatpak should be able to help. I hope this info is helpful.

**Reply by curtisy1** [reply]
Thank you! I'll have a look tomorrow and see if I can come up with something that works
